service/dap/server_test.go

```diff
@@ -2172,6 +2196,8 @@ func testStepParkedHelper(t *testing.T, client *daptest.Client, fixture protest.
 	return goroutineId
 }
 
+// TestStepOutPreservesGoroutine is inspired by proc_test.TestStepOutPreservesGoroutine
+// and checks that StepOut preserves the currently selected goroutine.
```

Comment:
This is a diff from https://github.com/go-delve/delve/pull/2417. I did not merge for the merge to show up as its own commit. Oops.